### ISSUE TITLE
python-chaco is broken in Ubuntu... 

### ISSUE THREAD

**Issue by gerhardt** [enthought/chaco#273]
The image plots are not working. This prevents us from distributing our code, which requires this feature. It's especially annoying since this affects all Ubuntu users for quite some time now.

Please see

https://bugs.launchpad.net/ubuntu/+source/python-chaco/+bug/1145575?comments=all

Best Regards

                                  Ilja

**Comment by corranwebster** [comment]
This looks like an issue in pyface and/or Wx.  Which version of WxPython are you using?

**Comment by gerhardt** [comment]
On 06/05/2015 10:49 AM, Corran Webster wrote:
> This looks like an issue in pyface and/or Wx. Which version of WxPython
> are you using?

It's python-wxgtk2.8 = 2.8.12.1+dfsg-2ubuntu2 (standard with 14.04) -
but, I suspect, that this bug might be related to the kiva bug /
python-enable since:

https://github.com/enthought/enable/issues/74
https://github.com/enthought/enable/issues/97

and further:

strace python
/usr/share/doc/python-chaco/examples/demo/advanced/scalar_image_function_inspector.py

ends with s.th. like this:

[...]
recvmsg(6, 0x7fffb532db60, 0)           = -1 EAGAIN (Resource
temporarily unavailable)
recvmsg(6, 0x7fffb532db60, 0)           = -1 EAGAIN (Resource
temporarily unavailable)
mmap(NULL, 1929216, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1,
0) = 0x7fcccf5c8000
stat("/usr/lib/python2.7/dist-packages/kiva/fonttools/numpy",
0x7fffb532b600) = -1 ENOENT (No such file or directory)
open("/usr/lib/python2.7/dist-packages/kiva/fonttools/numpy.x86_64-linux-gnu.so",
O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib/python2.7/dist-packages/kiva/fonttools/numpy.so",
O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib/python2.7/dist-packages/kiva/fonttools/numpymodule.so",
O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib/python2.7/dist-packages/kiva/fonttools/numpy.py",
O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib/python2.7/dist-packages/kiva/fonttools/numpy.pyc",
O_RDONLY) = -1 ENOENT (No such file or directory)
stat("/usr/share/fonts/truetype/ttf-bitstream-vera/Vera.ttf",
{st_mode=S_IFREG|0644, st_size=65932, ...}) = 0
open("/usr/share/fonts/truetype/ttf-bitstream-vera/Vera.ttf", O_RDONLY) = 7
fcntl(7, F_SETFD, FD_CLOEXEC)           = 0
fstat(7, {st_mode=S_IFREG|0644, st_size=65932, ...}) = 0
mmap(NULL, 65932, PROT_READ, MAP_PRIVATE, 7, 0) = 0x7fcce22a6000
close(7)                                = 0
brk(0x2f77000)                          = 0x2f77000
brk(0x2fcd000)                          = 0x2fcd000
brk(0x306a000)                          = 0x306a000
brk(0x3106000)                          = 0x3106000
brk(0x3128000)                          = 0x3128000
mmap(NULL, 262144, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1,
0) = 0x7fccd801a000
mmap(NULL, 262144, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1,
0) = 0x7fcccf588000
munmap(0x7fccd801a000, 262144)          = 0
brk(0x31bc000)                          = 0x31bc000
brk(0x3258000)                          = 0x3258000
stat("/usr/share/fonts/truetype/ttf-bitstream-vera/Vera.ttf",
{st_mode=S_IFREG|0644, st_size=65932, ...}) = 0

[...]

> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/enthought/chaco/issues/273#issuecomment-109205784>.
> 


-- 
Dr. Ilja Gerhardt :: MPI Stuttgart :: Snail-Mail: University of
Stuttgart :: 3. Institute of Physics :: Room 6.549 :: Pfaffenwaldring 57
:: 70569 Stuttgart :: Tel: +49-711-685-65283 :: ilja@quantumlah.org ::
http://gerhardt.ch

**Comment by corranwebster** [comment]
Yes, that would indeed seem likely.  I'm not sure what the status is with the Enable issue.

**Comment by gerhardt** [comment]
Hi Corran,

do you have any clue who would might feel responsible (and capable) to
address the python-enable bug? Presently, we're writing a piece of
software, and we have no good clue how to deal with this broken
density-plot issue. With Mac and Windoze it's all fine, all Ubuntu
versions segfault... When it's tunneling upstream is another question,
indeed.

Cheers!

                    Ilja

On 06/05/2015 04:44 PM, Corran Webster wrote:
> Yes, that would indeed seem likely. I'm not sure what the status is with
> the Enable issue.
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/enthought/chaco/issues/273#issuecomment-109315539>.
> 


-- 
Dr. Ilja Gerhardt :: MPI Stuttgart :: Snail-Mail: University of
Stuttgart :: 3. Institute of Physics :: Room 6.549 :: Pfaffenwaldring 57
:: 70569 Stuttgart :: Tel: +49-711-685-65283 :: ilja@quantumlah.org ::
http://gerhardt.ch

**Comment by corranwebster** [comment]
In one of the Enable issues, @pankajp mentions some ways to avoid the issue.  Which of these is feasible depends on your distribution situation.  The one that you can control best is to build your own Enable which works around the bug and distribute that with your application.

If that isn't practical, it may be possible to get the maintainers of the Ubuntu Enable package to build with optimizations turned off, which would also fix the issue for you.

I don't have time to look into what would be needed to change the code which is having trouble, but if someone were to submit a patch which gets around the GCC bug, it would likely quickly be accepted, and quickly put out a bugfix release for it.

**Comment by ptomato** [comment]
Here's a link to the Ubuntu issue, where I suggested turning off optimization for the Enable package: https://bugs.launchpad.net/ubuntu/+source/python-chaco/+bug/1145575
